bamCoverage --extendReads crashes when no default fragment size is provided

I have found `bamCoverage` crashes sometimes on the `scale_bigwig` step. 

It seems to be happening when `--extendReads` is provided with no default fragment size.

I think it does not come from bam file manipulation in the duplicate marking, because I have tried to run the same deepTools command with the `mapped` equivalent bam file (directly coming from bowtie) and it still crashes.

The error I get is this one from deepTools:

    AssertionError: fragment start greater than fragmentend for read NB501365:478:H2HFNBGXF:2:22301:2673:4193_CATGTA

However, if I go and look at the BAM file:

    NB501365:478:H2HFNBGXF:2:12207:13587:20077_TGACTA	163	chr7	125003593	42	34M	=	125003631	74	TTCTACTGTCTGAAAGGATCATAATGACAAAAAT	/AA/A/E6EEEEEEEEEEEEE/EEEEEEEEEEEE	AS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:34	YS:i:0	YT:Z:CP
    NB501365:478:H2HFNBGXF:2:12207:13587:20077_TGACTA	83	chr7	125003631	42	36M	=	125003593	-74	GCTACCAATAATATGGGAAAATATTATTTGTTAGAA	E/EEE<EA6/EEEA/EE/</EE/EEEEEEAAE//EE	AS:i:0	XN:i:0	XM:i:0	XO:i:0	XG:i:0	NM:i:0	MD:Z:36	YS:i:0	YT:Z:CP

The corresponding lines are perfectly valid, even though they have some peculiarities: First mate coordinates (second in line, because it's on the minus strand, according to the 83 flag field) are indeed less than second mate (125003593).

However I don't think this is the real problem because it seems like a fairly usual situation and some of the scaled bw files are fully produced. I also found one where the paired mate is not mapped and also crashes:

    NB501365:478:H2HFNBGXF:2:23111:15285:18671_AAACTT       137     chr17   72329   1       34M     =       72329   0       ACTGAGAAAGTTCAGAGCCTGGAGAGAAGGCTGG    6AAAAEEEEEEEEEEEEEAEEEEEEEAAE/EEEE      AS:i:0  XN:i:0  XM:i:0  XO:i:0  XG:i:0  NM:i:0  MD:Z:3 4       YT:Z:UP
    NB501365:478:H2HFNBGXF:2:23111:15285:18671_AAACTT       69      chr17   72329   0       *       =       72329   0       ACGACGCTCTTCCGATCTGAAGGGGCACATCTACAGC AAEEEEEEEAEEEEAEEAEEEEEEEE/EEEEE/EEEA   YT:Z:UP


I have also tried to isolate reads that cause this error and look at them on IGV, they also load fine and make sense when I look at them.

The same `bamCoverage` run does not crash if I do provide a fragment size value:

    bamCoverage -p 20 --binSize 1 --extendReads 150 --scaleFactor 69.91912350319042 --bam mapped/bamfile.bam -o scaled/scaled.bw

However I am not sure of the effect this has in the final bigwig file, so I am not sure if this is a fix.

Another thing is that if I run several times with the same BAM file, I get it crashing in different read IDs if `-p` > 1. This I guess it is due to parallelization of the process, but it suggests that this is a frequent enough problem to keep drawing different reads that make it crash.

The TLEN field is also a bit suspicious in those cases but I need to look into it in more detail.

--- a/Snakefile
+++ b/Snakefile
@@ -321,7 +321,7 @@ rule scaled_bigwig:
         factor="factors/{library}.factor.txt",
         fragsize="restricted/{library}.fragsize.txt",
         bam="restricted/{library}.bam",
-        bai="restricted/{library}.bai"
+        bai="restricted/{library}.bai",
     threads: 20
     shell:
         # TODO also run this
@@ -334,7 +334,7 @@ rule scaled_bigwig:
         " --scaleFactor $(< {input.factor})"
         " --bam {input.bam}"
         " -o {output.bw}"
- 
+
 
 rule stats:
     output:

